pkgs/applications/editors/neovim/build-neovim-plugin.nix

```diff
@@ -13,11 +13,12 @@ in
   {
     # the lua attribute name that matches this vim plugin. Both should be equal
     # in the majority of cases but we make it possible to have different attribute names
-    luaAttr ? (normalizeName attrs.pname)
+    luaAttr ? (lua.pkgs.${normalizeName attrs.pname})
     , ...
   }@attrs:
     let
-      originalLuaDrv = lua.pkgs.${luaAttr};
+      # warning added Sep 2024
+      originalLuaDrv = lib.warnIf (lib.typeOf luaAttr == "string") "luaAttr as string is deprecated. Pass a lua derivation directly ( e.g., `buildNeovimPlugin { luaAttr = lua.pkgs.plenary-nvim; }`)" luaAttr;
```

Comment:
issue: a deprecation causes a warning to be printed, but still works. 
This is a breaking change, because passing in a string is no longer possible. 

This should either use `lib.throwIf` (with the wording "no longer supported" instead of "deprecated" or it should use `lib.warnIf` and fall back to `lua.pkgs.${luaAttr}`.

Reply:
indeed. I've converted it into an if/else in a followup commit and tested with luaAttr = "haskell-tools.nvim". If that's good for you. I'll squash on merge .

Reply:
I had to restore the default for luaAttr else the plugins in `vim/generated.nix` would fail to eval. So if you dont set pname and luaAttr the eval will fail. I preferred to do that since the handling of vim/generated.nix will change one way or another.
(my first take is https://github.com/NixOS/nixpkgs/pull/342884 but I dont like adding yet another updater so hopefully we can do better)